pkg/machinery/config/schemas/config.schema.json

```diff
@@ -0,0 +1,3212 @@
+{
```

Comment:
This is the new schema. Main changes are:
- Types in "defs" are now prefixed with their package: `AdminKubeconfigConfig` -> `v1alpha1.AdminKubeconfigConfig`, to ensure they stay unique in the multidoc world.
- As the root property, use `oneOf` instead of `$ref` (This schema will validate a single config doc at a time).

There are also some minor fixes in the schema since we last visited it.